packages/launchpad/src/Main.vue

```diff
@@ -193,6 +193,50 @@ const resetErrorAndLoadConfig = (id: string) => {
 }
 const query = useQuery({ query: MainLaunchpadQueryDocument })
 const currentProject = computed(() => query.data.value?.currentProject)
+const hasBaseError = computed(() => !!query.data.value?.baseError)
+
+const refetchDelaying = ref(false)
+const refetchCount = ref(0)
+
+/*
+ * Sometimes the config file has not been loaded by the DataConntext's config manager by the
+ * time the MainLaunchpadQueryDocument request is sent off. The server ends up resolving
+ * the isLoadingConfigFile field as false. In certain situations, there can be a race between
+ * opening the project and the DataContext completing its retrieval of the configuration.
+ * In these cases, we want to retry the query until the config file is fully loaded.
+ *
+ * If the ProjectConfigIPC encounteres an error while loading the config, it will update the
+ * baseError field via subscription, so there is not a limit set here on retries.
+ */
+
+watch(
+  [currentProject, query.fetching],
+  ([currentProject, isFetchingProject]) => {
+    const isLoadingConfig = currentProject?.isLoadingConfigFile
+
+    /*
+     * conditions for refetch are:
+     * - There is a current project, but Config file has not yet loaded
+     * - There are no pending (delayed) refetches, or fetches in progress
+     * - There is no baseError - we don't want to continue to refetch if
```

Comment:
Do we want to have a test for the error scenario?

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/29597/commits/c15a55d4d1cf2886f903c4c17a5282db2f9e4243 - thanks, it found a bug!